cmd/otel-allocator/main.go

```diff
@@ -111,6 +111,17 @@ func main() {
 			setupLog.Error(err, "Can't start the prometheus watcher")
 			os.Exit(1)
 		}
+		// apply the initial configuration
+		promConfig, loadErr := promWatcher.LoadConfig(ctx)
+		if loadErr != nil {
+			setupLog.Error(err, "Can't load initial Prometheus configuration from Prometheus CRs")
+			os.Exit(1)
+		}
+		loadErr = targetDiscoverer.ApplyConfig(allocatorWatcher.EventSourcePrometheusCR, promConfig.ScrapeConfigs)
```

Comment:
i.e. it's fine to do the load config here, but couldn't we just set it as cfg.PromConfig and let the rungroup handle the call?

Reply:
We can't in general, because `cfg.PromConfig` might not be empty. I could move it into the rungroup which established the Prometheus CR Watcher, if you think that'd be cleaner.

Reply:
The calls look similar, but they actually refer to different event sources.